src/components/post-details-page/PostDetailsPage.js

```diff
@@ -25,22 +31,54 @@ class PostDetailsPage extends Component {
     //  результат выполнения запроса нужно положить в стейт в post
     //  когда запрос выполнится - не забудьте поменять индикатор загрузки isLoading на false
     //  обратите внимание, что результат выполнения запроса - ОБЪЕКТ, а не массив
+    const  { match: {params:{id}} } = this.props;
+    this.setState({
+      isLoading: true
+    });
+
+    let response = await fetch(`https://gorest.co.in/public-api/posts/${id}?access-token=${accessToken}`);
+
+    if (response.ok) {
+      let json = await response.json();
+
+      const { result } = json;
+
+      if (typeof (result) === "object") { // во время выполнения запроса м.б. вариант когда result не массив
+        this.setState({
+          isLoading: false,
+          commentsLoaded: true,
+          error: '',
+          posts: result || undefined // изменена проверка, если results существовать не будет - закидываем пустой массив
```

Comment:
in render your are using `posts.map` so you can't put here `undefined` as default value - should be empty array